src/ORM/Query.php

```diff
@@ -203,7 +202,7 @@ public function select($fields = [], $overwrite = false)
      *
      * This method returns the same query object for chaining.
      *
-     * @param \Cake\ORM\Table $table The table to pull types from
+     * @param \Cake\Datasource\RepositoryInterface $table The table to pull types from
      * @return $this
      */
     public function addDefaultTypes(Table $table)
```

Comment:
@thinkingmedia Then we have a different problem here.
The typehint going in is of the interface type, so the IDE will complain that this is technically not quite correct, probably should be using more interface segregation in the future here and then we can always use interface typehints.

Anyway, I can then also revert the doc block change for this method and we keep the current warning.